contentScript.js

```diff
@@ -5,33 +5,27 @@ chrome.runtime.onMessage.addListener(
   function(request, sender, sendResponse) {
 
     if(request.parse ){
-      
+
       var wordlist = request.wordlist;
       var entries =  document.querySelectorAll("[class^='athing']");
 
       var entryParent = null;
 
+      var re = new RegExp('(' + wordlist.join('|') + ')', 'gi');
```

Comment:
only need to declare the regex once before the we loop over the entries, no more need for nested FOR loops.

there's only 30 entries per page, and typically i wouldn't expect most users to add more than a handful of terms so I think the regex will perform ok.